File: desktop/src/main/java/com/techfinance/pessoal/desktop/service/AuthService.java
package com.techfinance.pessoal.desktop.service;

import com.google.inject.Inject;
import com.google.inject.Singleton;
import com.techfinance.pessoal.desktop.dto.request.LoginRequest;
import com.techfinance.pessoal.desktop.dto.request.RegisterRequest;
import com.techfinance.pessoal.desktop.dto.response.AuthResponse;
import com.techfinance.pessoal.desktop.infrastructure.http.ApiClient;
import com.techfinance.pessoal.desktop.infrastructure.session.SessionManager;

@Singleton
public class AuthService {

    private final ApiClient apiClient;
    private final SessionManager session;
    private final UserApiService userApiService;

    @Inject
    public AuthService(
            ApiClient apiClient,
            SessionManager session,
            UserApiService userApiService) {

        this.apiClient = apiClient;
        this.session = session;
        this.userApiService = userApiService;
    }

    public void register(RegisterRequest request) {
        AuthResponse response = apiClient.postPublic(
            "/auth/register",
            request,
            AuthResponse.class
        );

        establishSession(response);
    }

    public void login(LoginRequest request) {
        AuthResponse response = apiClient.postPublic(
            "/auth/login",
            request,
            AuthResponse.class
        );

        establishSession(response);
    }

    public void logout() {
        session.clear();
    }

    public boolean isAuthenticated() {
        return session.isAuthenticated();
    }

    public String getUsername() {
        return session.getUsername();
    }

    public String getName() {
        return session.getName();
    }

    public String getToken() {
        return session.getToken();
    }

    private void establishSession(AuthResponse response) {
        session.setToken(response.token());
        session.setUsername(response.username());
        session.setName(response.name());

        var user = userApiService.findByUsername(response.username());
        session.setUserId(user.id());
    }
}

File: web/src/services/authService.ts
import { apiClient } from '../infrastructure/http/apiClient'
import type { LoginRequest, RegisterRequest, AuthResponse } from '../types/auth'
import {
  clearSession,
  getSession,
  isAuthenticated,
  setSession,
} from '../infrastructure/session/sessionManager'
import { userApiService } from './userApiService'

class AuthService {
  async register(request: RegisterRequest): Promise<void> {
    const response = await apiClient.postPublic<AuthResponse>(
      '/auth/register',
      request,
    )

    await this.establishSession(response)
  }

  async login(request: LoginRequest): Promise<void> {
    const response = await apiClient.postPublic<AuthResponse>(
      '/auth/login',
      request,
    )

    await this.establishSession(response)
  }

  logout(): void {
    clearSession()
  }

  isAuthenticated(): boolean {
    return isAuthenticated()
  }

  getUsername(): string | null {
    return getSession()?.username ?? null
  }

  getName(): string | null {
    return getSession()?.name ?? null
  }

  getToken(): string | null {
    return getSession()?.token ?? null
  }

  private async establishSession(response: AuthResponse): Promise<void> {
    setSession({
      token: response.token,
      username: response.usuario,
      name: response.nome,
      userId: '',
    })

    const user = await userApiService.findByUsername(response.usuario)

    setSession({
      token: response.token,
      username: response.usuario,
      name: response.nome,
      userId: user.id,
    })
  }
}

export const authService = new AuthService()
